docs/DevGuide/TensorExpressions.md

```diff
@@ -0,0 +1,411 @@
+\ref tnsr "tnsr" type aliases.
+
+# Syntax {#te_syntax}
+`TensorExpression`s are arithmetic expressions of `Tensor`s that can be
+evaluated using `tenex::evaluate`. Terms used in the expression may be `Tensor`s
+or numbers (see [supported types](#te_data_type_support)).
+
+As a simple example of how `TensorExpression`s are used, if you would like to
+raise the index of some `Tensor` `R` with some spacetime metric `Tensor` `g`,
+i.e. \f$R^c{}_b = R_{ab} g^{ac}\f$, you can compute this with
+`TensorExpression`s by doing:
+```
+auto R_up = tenex::evaluate<ti::C, ti::b>(R(ti::a, ti::b) * g(ti::A, ti::C));
+```
+
+where `R_up`, `R`, and `g` are rank 2 spacetime `Tensor`s and the `ti::*`
+variables are `TensorIndex`s. Here, the argument to
+\ref tenex::evaluate "evaluate" is the RHS tensor expression to compute, `R_up`
+is the result LHS `Tensor`, and the template arguments to
+\ref tenex::evaluate "evaluate" are the LHS `Tensor`'s indices. The LHS symmetry
+will be deduced from the RHS tensors' symmetries and order of operations.
+
+Alternatively, if you already have a LHS `Tensor` variable, you can pass it into
+the following \ref tenex::evaluate "evaluate" overload, where the LHS `Tensor`
+provided will be assigned to the result of the RHS expression:
+
+```
+tenex::evaluate<ti::C, ti::b>(
+    make_not_null(&R_up), R(ti::a, ti::b) * g(ti::A, ti::C));
+```
+
+One advantage of this overload is that it uses the symmetry and index types
+(spatial/spacetime) of the provided LHS tensor instead of deducing them from the
+RHS expression, so this enables you to specify the LHS index structure in cases
+where the previous \ref tenex::evaluate "evaluate" overload does not deduce the
+one you want. Note that the LHS tensor does not need to be previously sized
+unless the data type is a Blaze vector type (e.g. `DataVector`) *and* the RHS
+expression contains no `Tensor` terms (see [example](#te_assigning_to_a_number)
+where sizing is necessary). This overload is also used to
+[assign subsets of tensor components](#te_assigning_subsets_of_components).
+
+## Tensor indices (TensorIndexs) {#te_tensor_indices}
+
+`TensorIndex`s represent generic tensor indices and are supplied as
+comma-separated lists in two places: (1) in parentheses for each tensor in the
+RHS expression and (2) in the template parameters of
+\ref tenex::evaluate "evaluate" to specify the order of the LHS result tensor's
+indices.
+
+Each `TensorIndex` takes the form `ti::*` where `*` is a letter that encodes
+index properties:
+- Uppercase letters denote upper indices and lowercase letters denote lower
+indices
+- Letters `A/a - H/h` indicate spacetime indices, `I/i - N/n` indicate spatial
+indices, and `T/t` indicates a concrete time index. This is what is currently
+defined, but more spatial and spacetime indices (letters) can easily be added if
+needed
+
+The properties of each `TensorIndex` and the `Tensor`'s indices (typelist of
+\ref SpacetimeIndex "TensorIndexType"s) must be compatible:
```

Comment:
Is there anything special about certain `tensorIndex`s (other than `T/t`)? Like say I want to contract a `tnsr::ab`. Do I have to do `tenex::evaluate(R(ti::A, ti::a))`? Or can I do `tenex::evaluate(R(ti::H, ti::h))`? Similarly with result tensors: `auto L = tenex::evaluate(R(ti::a, ti::b) + S(ti::a, ti::b))` vs `auto L = tenex::evaluate(R(ti::e, ti::f) + S(ti::e, ti::f))`?

If we have to use `A/a` first, that should be noted. If not, say that it doesn't matter, but we still tend to use `A/a` and `I/i` first and then move down the alphabet.

Reply:
No, `T/t` is the only special one. `a`, `b`, `c`, ..., `h` are equivalent and it doesn't matter what you use. I used a lot of `a` and `b` just because, but if you think the guide made it seem like you had to use them with a certain precedence, then I can either add a short line somewhere saying it doesn't matter or I can vary the letters I use throughout the guide. Let me know which you prefer!

Reply:
Oh sorry, reread your comment and it answered my question - I'll just say explicitly that they are equal and there is no precedence